src/builtins/impls/urlquery.rs

```diff
@@ -16,28 +16,70 @@
 
 use std::collections::HashMap;
 
-use anyhow::{bail, Result};
+use serde_json::Value;
+use url::form_urlencoded::{byte_serialize, parse};
+
+const QUERY_CONCAT_CHAR: &str = "&";
 
 /// Decodes a URL-encoded input string.
-#[tracing::instrument(name = "urlquery.decode", err)]
-pub fn decode(x: String) -> Result<String> {
-    bail!("not implemented");
+#[tracing::instrument(name = "urlquery.decode")]
+pub fn decode(x: String) -> String {
```

Comment:
This looked weird to me on how it seemed to work. I dug a bit in the implementation of `form_urlencoded::parse`, and it seems it really is more for url-encoded key-value pairs like `a=b&c=d`. So it indeed does not output what's expected:

```rego
package test

a := urlquery.decode("&&&&")
b := urlquery.decode("====")
c := urlquery.decode("&=&=")
```

outputs

```json
{
  "a": "",
  "b": "===",
  "c": ""
}
```

and with OPA directly:

```json
{
    "a": "&&&&",
    "b": "====",
    "c": "&=&="
}
```

I think what would be best would be to leverage a crate like [`urlencoding`](https://lib.rs/crates/urlencoding) for this?

Reply:
yes, did't thought to check those scenarios, 
I changed the implementation + added them to the test plan